meta-dts-distro/recipes-dts/dts/dts/dts-functions.sh

```diff
@@ -967,3 +989,122 @@ set_intel_regions_update_params() {
+          echo
+          break
+          ;;
+        n|N)
+          compare_versions $DASHARO_VERSION $UPDATE_VERSION
+          if [ $? -ne 1 ]; then
+            error_exit "No update available for your machine"
+          fi
+          echo "Will not install Dasharo heads firmware. Proceeding with regular Dasharo firmware update."
+          break
+          ;;
+        *)
+          ;;
+      esac
+    done
+  elif [ -v DES_IS_LOGGED ] && [ -v HEADS_LINK_DES ]; then
+    [ "$DASHARO_FLAVOR" == "Dasharo (coreboot+heads)" ]
+    local _heads_des=1
+    curl -sfI -u "$USER_DETAILS" -H "$CLOUD_REQUEST" "$HEADS_LINK_DES" -o /dev/null
+    _heads_des=$?
+    # We are on heads, offer switch back or perform update if DES gives access to heads
+    if [ "$DASHARO_FLAVOR" == "Dasharo (coreboot+heads)" ]; then
+      while : ; do
+        echo
+        read -r -p "Would you like to switch back to the regular Dasharo firmware? (Y|n) " OPTION
+        echo
+
+        case ${OPTION} in
+          yes|y|Y|Yes|YES)
+            echo
+            echo "Switching back to regular Dasharo firmware v$UPDATE_VERSION"
+            echo
+            FLASHROM_ADD_OPT_UPDATE_OVERRIDE="--ifd -i bios"
+            break
+            ;;
+          n|N)
+            if [ $_heads_des -ne 0 ]; then
+              error_exit "No update available for your machine"
+            fi
+            UPDATE_VERSION=$HEADS_REL_VER_DES
+            compare_versions $DASHARO_VERSION $UPDATE_VERSION
+            if [ $? -ne 1 ]; then
+              error_exit "No update available for your machine"
+            fi
+            echo "Will not switch back to regular Dasharo firmware. Proceeding with Dasharo heads firmware update to $UPDATE_VERSION."
+            FLASHROM_ADD_OPT_UPDATE_OVERRIDE="--ifd -i bios"
+            BIOS_HASH_LINK="${HEADS_LINK_DES}.sha256"
+            BIOS_SIGN_LINK="${HEADS_LINK_DES}.sha256.sig"
+            BIOS_LINK=$HEADS_LINK_DES
+            break
+            ;;
+          *)
+            ;;
+        esac
+      done
+    fi
+  elif [ ! -v DES_IS_LOGGED ] && [ "$DASHARO_FLAVOR" == "Dasharo (coreboot+heads)" ]; then
+    # Not logged with DES and we are on heads, offer switch back
+    while : ; do
+      compare_versions $DASHARO_VERSION $HEADS_REL_VER_DES
+      if [ $? -eq 1 ]; then
+        print_warning "You are running heads firmware, but did not provide DES credentials."
+        print_warning "There are updates available if you provide DES credentials in main DTS menu."
+      fi
+      echo
+      read -r -p "Would you like to switch back to the regular Dasharo firmware? (Y|n) " OPTION
+      echo
+
+      case ${OPTION} in
+        yes|y|Y|Yes|YES)
+          echo
+          echo "Switching back to regular Dasharo firmware v$UPDATE_VERSION"
+          echo
+          FLASHROM_ADD_OPT_UPDATE_OVERRIDE="--ifd -i bios"
+          break
+          ;;
+        n|N)
+          compare_versions $DASHARO_VERSION $HEADS_REL_VER_DES
```

Comment:
@miczyg1 do we need to compare this one more time if we do that at the beginning of this while loop?

Reply:
I think I messed up a little here. We should simply skip the compare and the IF below it. I will also move the 
`echo "Latest available Dasharo version: $HEADS_REL_VER_DES"`  to be printed before the loop along with the warning about credentials and possible updates...

Reply:
https://github.com/Dasharo/meta-dts/pull/95/commits/d1369f5f4ead01bf0f052269f75caff3cf852a68